### PR TITLE
apis: add default PriorityClass and QoSClass, revise batch resource calculation

### LINKED ISSUE #1441: [proposal] Revise batch resource calculation
**What is your proposal**:

The batch resource calculation in the koord-manager should be revised.
Besides the QoS class of pods, the calculator should also check their priority classes.
Note that a pod without a priority class is regarded as Prod by default, and a pod without a QoS class is regarded as LS by default.

**Why is this needed**:

Since the Mid-tier is supported, there can be BE pods in different priority classes, where the requests and usages of Mid+BE pods should be counted as the high-priority and the batch resource should reclaim the unused part.

**Is there a suggested solution, if so, please add it**:

Revise the BatchResource plugin's implementation in the koord-manager.

### PR DESCRIPTION
**Pull request by saintube**
…yclass and qosclass

### Ⅰ. Describe what this PR does

- koord-manager: revise batch resource calculation, identifying the pods' priorityclasses.
- koord-manager: fix a bug where the `noderesource-plugin` arg allows no plugin by default.
- util: add default PriorityClass and QoSClass when any of them is missing.

### Ⅱ. Does this pull request fix one issue?

Fixes #1441.

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: pkg/util/slo_utils.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	DefaultQoSClass      = extension.QoSLS
+)
+
+var (
+	// DefaultPriorityClassWithQoS is the default PriorityClass according to its QoSClass when the pod does not specify
+	// a PriorityClass explicitly.
+	// NOTE: The pod with PriorityClassNone is considered as PriorityClassProd, and the pod with QoSNone is considered
+	// as qos=LS. A special case is that the pod with PriorityClassNone+QoSBE is considered as PriorityBatch to get
+	// compatible to with existing BE pods.
+	DefaultPriorityClassWithQoS = map[extension.QoSClass]extension.PriorityClass{
```
Comment (eahydra) [inline pkg/util/slo_utils.go:38]: Why not move the file to package "apis/extensions"? Should we replace all code that cares about PriorityClass and QoS ?

File: pkg/util/slo_utils.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+)
+
+// NOTE: functions in this file can be overwritten for extension
+
+const (
+	DefaultPriorityClass = extension.PriorityProd
```
Comment (eahydra) [inline pkg/util/slo_utils.go:28]: In package `apis` already has the same definition but DefaultPriorityClass is PriorityNone.
The new definition looks to change the default behavior of Koordinator models.

Comment (saintube) [reply pkg/util/slo_utils.go]: done

File: pkg/util/slo_utils.go (hunk 1 above)
Comment (saintube) [reply pkg/util/slo_utils.go]: OK

**Review by eahydra — COMMENTED** [review]
/lgtm

File: apis/extension/priority_utils.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+// NOTE: functions in this file can be overwritten for extension
+
+var (
+	DefaultPriorityClass = PriorityNone
+
+	// DefaultPriorityClassWithQoS is the default PriorityClass according to its QoSClass when the pod does not specify
+	// a PriorityClass explicitly.
+	DefaultPriorityClassWithQoS = map[QoSClass]PriorityClass{
+		QoSLSE:  PriorityProd,
```
Comment (zwzhang0107) [inline apis/extension/priority_utils.go:27]: QoSSystem

Comment (saintube) [reply apis/extension/priority_utils.go]: fixed

File: apis/extension/priority.go [hunk 4]
```diff
@@ -52,6 +48,16 @@ var (
...
 )
 
+// GetPodPriorityClassWithDefault gets the pod's PriorityClass with the default config.
+func GetPodPriorityClassWithDefault(pod *corev1.Pod) PriorityClass {
+	priorityClass := GetPriorityClass(pod)
+	if priorityClass != PriorityNone {
+		return priorityClass
+	}
+
+	return DefaultPriorityClassWithQoS[GetPodQoSClass(pod)]
```
Comment (hormes) [inline apis/extension/priority.go:58]: 现在 QoS 也缺少兼容逻辑，如果用啥啥都没填，应该兼容 k8s qos

**Review by eahydra — COMMENTED** [review]
/lgtm

File: apis/extension/priority.go (hunk 4 above)
Comment (saintube) [reply apis/extension/priority.go]: Done

**Review by hormes — APPROVED** [review]

**Pull request merged**